### ISSUE TITLE
test for known example

### ISSUE THREAD

**Issue by Sandy4321** [dstein64/PyFactorizationMachines#3]
may you please verify  how it works for 
example from 
http://srome.github.io/Leveraging-Factorization-Machines-for-Sparse-Data-and-Supervised-Visualization/
data size is 
X_train.shape
(7580, 1048576)
y_train.shape
(7580,)

 I send you customized code from this link

#S_May2_hashing_FM_sparse.py
#S_apr25_FM_sparse
#http://srome.github.io/Leveraging-Factorization-Machines-for-Sparse-Data-and-Supervised-Visualization/

import pickle
import pandas as pd
from sklearn.feature_extraction import FeatureHasher
#original from PyFactorizationMachines.src.pyfm import FactorizationMachineClassifier
from pyfms import Classifier
from sklearn.model_selection import train_test_split

if 1:
    from sklearn.datasets import fetch_20newsgroups
    twenty_train = fetch_20newsgroups(subset='train', 
                                      shuffle=True, 
                                      random_state=42)
    
    
    
    filehandler = open(b"twenty_data.pkl","wb")
    pickle.dump(twenty_train,filehandler)
else:

    file_name = open("twenty_data.pkl",'rb')
    twenty_train= pickle.load(file_name)
twenty_train.target_names

def define_label(x, target_names):
    name = target_names[x]
    return 1 if 'comp' in name else 0
    
target = pd.Series(twenty_train.target).apply(lambda x : define_label(x,twenty_train.target_names))
target.mean() # Prevalence of the label

def clean_text(text):
    # Basic cleaning
    text=text.replace('\n','').replace('\t','').replace('<','').replace('>','').replace('|','')
    return [x for x in text.split(' ') if len(x) > 3]

X = [clean_text(x) for x in twenty_train.data]

# Hash away!
fh = FeatureHasher(input_type='string',non_negative=True)
X_t = fh.transform(X)

# Bin the inputs so that the "interaction" terms are more interpretable
X_bin = X_t.copy()
X_bin[X_bin >= 1] = 1

f = Classifier(X_t.shape[1] , k=2, X_format="csr")
X_train, X_test, y_train, y_test = train_test_split(X_bin, target, test_size=0.33, random_state=42)

f.fit(X_train, y_train,  verbosity=50, nb_epoch=200)

q=4

**Comment by Sandy4321** [comment]
is initial program output correct?
it takes for ever to calculate, interesting how fast is it on your computer ?

Microsoft Windows [Version 10.0.16299.371]
(c) 2017 Microsoft Corporation. All rights reserved.

E:\PyFactorizationMachines_May2\PyFactorizationMachines>python S_May2_hashing_FM_sparse.py
WARNING (theano.configdefaults): g++ not available, if using conda: `conda install m2w64-toolchain`
C:\Users\sndr\Anaconda3\lib\site-packages\theano\configdefaults.py:560: UserWarning: DeprecationWarning: there is no c++ compiler.This is deprecated and with Theano 0.11 a c++ compiler will be mandatory
  warnings.warn("DeprecationWarning: there is no c++ compiler."
WARNING (theano.configdefaults): g++ not detected ! Theano will be unable to execute optimized C-implementations (for both CPU and GPU) and will default to Python implementations. Performance will be severely degraded. To remove this warning, set Theano flags cxx to an empty string.
WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
C:\Users\sndr\Anaconda3\lib\site-packages\sklearn\feature_extraction\hashing.py:94: DeprecationWarning: the option non_negative=True has been deprecated in 0.19 and will be removed in version 0.21.
  " in version 0.21.", DeprecationWarning)
C:\Users\sndr\Anaconda3\lib\site-packages\pandas\core\series.py:696: FutureWarning:
Passing list-likes to .loc or [] with any missing label will raise
KeyError in the future, you can use .reindex() as an alternative.

See the documentation here:
http://pandas.pydata.org/pandas-docs/stable/indexing.html#deprecate-loc-reindex-listlike
  return self.loc[key]

**Comment by dstein64** [comment]
@Sandy4321, if you are encountering and issue, please be sure to provide the simplest possible example that reproduces the issue, so I can figure out what's wrong. The issue posted above is too large for me to be able to understand what the problem is, or debug.

Regarding the speed, please see my comment [here](https://github.com/dstein64/PyFactorizationMachines/issues/1#issuecomment-386158489), regarding the installation of `m2w64-toolchain`.
